mailet/standard/src/test/java/org/apache/james/transport/matchers/AtMostTest.java

```diff
@@ -56,103 +57,208 @@ private Mail createMail() throws MessagingException {
     void setup() throws MessagingException {
         this.matcher = new AtMost();
         FakeMatcherConfig matcherConfig = FakeMatcherConfig.builder()
-            .matcherName("AtMostFailureRetries")
-            .condition(CONDITION)
+            .matcherName("SingleMatcher")
+            .condition(TRIES_CONDITION)
             .build();
-
         matcher.init(matcherConfig);
     }
 
-    @Test
-    void shouldMatchWhenAttributeNotSet() throws MessagingException {
-        Mail mail = createMail();
-
-        Collection<MailAddress> actual = matcher.match(mail);
-
-        assertThat(actual).containsOnly(RECIPIENT1);
+    @Nested
+    class WrongConditionConfigurationTest {
```

Comment:
Test the following conditions:

```
abc:0
abc:-1
abc:def
```

Reply:
what about:
`:-1`
`abc: `
` : `
`abc`

Reply:
Add them too ^^